nncf/onnx/graph/transformations/commands.py

```diff
@@ -22,18 +22,26 @@
 
 
 class ONNXTargetPoint(TargetPoint):
-    def __init__(self, target_type: TargetType, target_node_name: str, port_id: Optional[int] = None):
+    def __init__(
+        self,
+        target_type: TargetType,
+        target_node_name: str,
+        port_id: Optional[int] = None,
+        destination_node_names: Optional[List[str]] = None,
+    ):
         """
         Constructor.
 
         :param target_type: Target type of the target point.
         :param target_node_name: ONNX node name.
         :param port_id: Number of port id in case target_type is
             TargetType.PRE_LAYER_OPERATION or TargetType.POST_LAYER_OPERATION.
+        :param destination_node_names: List of successor the node names.
```

Comment:
I understand your need to store the affected node set in a `TargetPoint`, but the way you are doing it right now is too messy. Evidently we need at least a new `TargetType` here, something like `POST_BRANCH_WITH_PARTIAL_MERGE`, the additional field `branch_node_names` that would only be valid for this new type (can set up some runtime checks in `__init__` to make sure that we don't instantiate a wrong `TargetPoint`; by the way, is there an opportunity to do this in the base `TargetPoint` class, and maybe move `target_node_name` and `port_id` there too?). It will also then be possible to implement a check in the `PTTargetPoint` so that a target point with this type cannot be instantiated with a `RuntimeError("Torch does not support target type TargetType.POST_BRANCH_WITH_PARTIAL_MERGE")`, and maybe do the same for TF right now, but with a `NotImplementedError` instead.

Reply:
Added new options into the `TargetPoint`.
The new `TargetType.POST_BRANCH_WITH_PARTIAL_MERGE` looks too heavy in comparison to the existing ones, IMO.